src/PKSim.Presentation/UICommands/RenameSubjectUICommand.cs

```diff
@@ -1,44 +1,44 @@
-﻿using PKSim.Core.Model;
+﻿using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Data;
 using OSPSuite.Presentation.UICommands;
 using OSPSuite.Presentation.Views;
+using PKSim.Core.Model;
 
 namespace PKSim.Presentation.UICommands
 {
    public class RenameSubjectUICommand : ObjectUICommand<IMdiChildView>
    {
       private readonly RenameBuildingBlockUICommand _renameBuildingBlockCommand;
-      private readonly RenameObjectUICommand _renameSimulationComparisonCommand;
-      private readonly RenameObservedDataUICommand _renameObservedDataUICommand;
+      private readonly RenameObjectUICommand _renameObjectCommand;
+      private readonly RenameObservedDataUICommand _renameObservedDataCommand;
 
-      public RenameSubjectUICommand(RenameBuildingBlockUICommand renameBuildingBlockCommand, RenameObjectUICommand renameSimulationComparisonCommand, RenameObservedDataUICommand renameObservedDataUICommand)
+      public RenameSubjectUICommand(
+         RenameBuildingBlockUICommand renameBuildingBlockCommand, 
+         RenameObservedDataUICommand renameObservedDataCommand, 
+         RenameObjectUICommand renameObjectCommand)
       {
          _renameBuildingBlockCommand = renameBuildingBlockCommand;
-         _renameSimulationComparisonCommand = renameSimulationComparisonCommand;
-         _renameObservedDataUICommand = renameObservedDataUICommand;
+         _renameObjectCommand = renameObjectCommand;
+         _renameObservedDataCommand = renameObservedDataCommand;
       }
 
       protected override void PerformExecute()
       {
          var subject = Subject.Presenter.Subject;
          if (subject == null) return;
-         var buildingBlock = subject as IPKSimBuildingBlock;
-         if (buildingBlock != null)
-         {
-            execute(_renameBuildingBlockCommand, buildingBlock);
-            return;
-         }
 
-         var simulationComparison = subject as ISimulationComparison;
-         if (simulationComparison != null)
+         switch (subject)
          {
-            execute(_renameSimulationComparisonCommand, simulationComparison);
-            return;
+            case IPKSimBuildingBlock buildingBlock:
+               execute(_renameBuildingBlockCommand, buildingBlock);
+               break;
+            case DataRepository observedData:
+               execute(_renameObservedDataCommand, observedData);
+               break;
+            case IWithName withName:
+               execute(_renameObjectCommand, withName);
```

Comment:
this should be the standard for all object that are NOT buildign block and not observed data